src/platform/i386/chal/chal_prot.h

```diff
@@ -0,0 +1,94 @@
+/******************************************************************************
+Filename    : chal_prot.h
+Author      : Runyu Pan
+Date        : ?/?/2018
+Licence     : GPL v2; see COPYING for details.
+Description : The Composite chal general prototype interface.
+******************************************************************************/
+
+/* Begin Function Prototypes *************************************************/
+#ifndef CHAL_PROT_H
+#define CHAL_PROT_H
+
+/* Generic definitions */
+typedef enum {
+	PGTBL_PRESENT  = 1,
+	PGTBL_WRITABLE = 1 << 1,
+	PGTBL_USER     = 1 << 2,
```

Comment:
Couldn't these be `PGTBL_USER = X86_PGTBL_USER` (for example)?

Reply:
As I said elsewhere, this likely needs to be architecture specific, so consider that when interpreting this comment.